middleware/kubernetes/kubernetes.go

```diff
@@ -115,6 +138,73 @@ func (k *Kubernetes) recordsForTXT(r recordRequest) ([]msg.Service, error) {
 	return nil, nil
 }
 
+func (k *Kubernetes) CoreDNSRecord() nsRecord {
```

Comment:
Realistically/Practically, yes.  Which is why I treat it this way.

In theory, there could be multiple services pointing to the coredns pod, and those services may or may not have the DNS port open.  So a complete answer would be to look at each service that points to the pod, and examine the port forwarding in the service and pod configs to see which ones forward dns to coredns.

But in this PR, I am assuming there is one service for the DNS pod, and that service forwards DNS port to the pod.